pkg/tlsf/contrib/include/tlsf-malloc.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2014-2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup pkg_tlsf_malloc TLSF-based malloc.
+ * @ingroup  pkg
+ * @ingroup  sys
+ *
+ * @brief    TLSF-based global memory allocator.
+ *
+ * This is a malloc/free implementation built on top of the TLSF allocator.
+ * It defines a global tlsf_control block and performs allocations on that
+ * block. This implemetation replaces the system malloc
+ *
+ * Additionally, the calls to TLSF are wrapped in irq_disable()/irq_restore(),
+ * to make it thread-safe.
+ *
+ * If this module is used as the system memory allocator, then the global memory
+ * control block should be initialized as the first thing before the stdlib is
+ * used. Boards should use tlsf_add_global_pool() at startup to add all the memory
```

Comment:
In fact, I think this could be in auto_init modules. For the CCN example, this avoids the need to call tlsf_add_global_pool. If so, this function could be renamed to tlsf_malloc_init. What do you think?

Reply:
I'm not familiar with auto init. When initializing, tlsf_add_global_pool should be called with an address that depends on the memory map of the chip.

Reply:
yes, sure. Could be done in further PRs